packages/clarity-js/src/data/variable.ts

```diff
@@ -0,0 +1,35 @@
+import { Constant, Event, VariableData } from "@clarity-types/data";
+import encode from "./encode";
+
+export let data: VariableData = null;
+
+export function set(variable: string, value: string): void {
+    log(variable, value);
+    encode(Event.Variable);
+}
```

Comment:
do not encode individual calls - and instead use compute below for sending a single event per payload before upload. same for tags

Reply:
For tags we do want to encode individual calls so we send one tag per event. That's because time of the tag matters. In case of variables you are right and I've made the change.